rat/parser.py

```diff
@@ -585,26 +611,99 @@ def expression(self, min_precedence=0, is_lhs=False):
 
         return left
 
-    def statement(self):
+    def statement(self, fold=True):
         """
         Evaluates a single statement. Statements in rat are either assignments or sampling statements. They will get
         resolved into an `ops.Assignment` or an `ops.Distr` object.
         :return:
         """
-        self.check_bracket_stack()
+        return_statement = None
+
         token = self.peek()
         if Distributions.check(token):
             raise ParseError("Cannot assign to a distribution.", self.model_string, token.line_index, token.column_index)
 
         # Step 1. evaluate lhs, assume it's expression
         lhs = self.parse_nud(is_lhs=True)
-        # if isinstance(lhs, Param) or isinstance(lhs, Data):
         if isinstance(lhs, Expr):
             op = self.peek()
-            if AssignmentOps.check(op):
+            if op.value == "=" and self.peek(k=1).value == "match":
+                # Matched assignment block
+                reverse_order = False
+                self.remove()  # assignment operator
+                self.remove()  # "match"
+                self.expect_token(Identifier)
+                bounded_variable = self.peek()
+                if bounded_variable.value == "rev":
```

Comment:
Oh wow you already included this nice

Reply:
The sv rat file has the syntax specifications I based on.